WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductFromImage/AddProductFromImageViewModel.swift

```diff
@@ -42,15 +47,19 @@ final class AddProductFromImageViewModel: ObservableObject {
 
     private let addProductSource: AddProductCoordinator.Source
     private let onAddImage: (MediaPickingSource) async -> MediaPickerImage?
-    private var selectedImageSubscription: AnyCancellable?
+    private var subscriptions: Set<AnyCancellable> = []
 
     // MARK: - Scanned Texts
 
     @Published var scannedTexts: [ScannedTextViewModel] = []
+
+    @Published private var scannedTextValidation: [String: Bool] = [:]
+    @Published private(set) var regenerateButtonEnabled: Bool = false
+
     @Published private(set) var isGeneratingDetails: Bool = false
     @Published private(set) var errorMessage: String? = Localization.defaultError
 
-    private var selectedScannedTexts: [String] {
+    var selectedScannedTexts: [String] {
```

Comment:
❓ I think we don't have to remove the `private` keyword.

Reply:
Restored in 111a354ec4d39748a97768bafdf807b9f52d96e5.